File: swift-bridge/Sources/CoreLocationBridge/Floor.swift
import CoreLocation
import Foundation

func cl_floor_object(_ floor: CLFloor) -> [String: Any] {
    [
        "level": floor.level,
    ]
}

@available(macOS 12.0, *)
func cl_source_information_object(_ sourceInformation: CLLocationSourceInformation) -> [String: Any] {
    [
        "is_simulated_by_software": sourceInformation.isSimulatedBySoftware,
        "is_produced_by_accessory": sourceInformation.isProducedByAccessory,
    ]
}

File: src/floor.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
/// Snapshot of `CLFloor`.
pub struct Floor {
    /// Matches `CLFloor.level`.
    pub level: i64,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
/// Snapshot of `CLLocationSourceInformation`.
pub struct LocationSourceInformation {
    /// Matches `CLLocationSourceInformation.isSimulatedBySoftware`.
    pub is_simulated_by_software: bool,
    /// Matches `CLLocationSourceInformation.isProducedByAccessory`.
    pub is_produced_by_accessory: bool,
}
